microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/internal/jaxrs/java/JaxRsCodeLensParticipant.java

```diff
@@ -127,7 +135,9 @@ private static void collectURLCodeLenses(IJavaElement[] elements, String rootPat
 				if (isJaxRsRequestMethod(method) && Flags.isPublic(method.getFlags())) {
 					String baseURL = params.getLocalBaseURL();
 					String openURICommandId = params.getOpenURICommand();
-					CodeLens lens = createURLCodeLens(baseURL, rootPath, openURICommandId, (IMethod) element, utils);
+					CodeLens lens = rootPathProperty != null ? 

```

Comment:
Keep existing code:

`CodeLens lens = createURLCodeLens(baseURL, rootPath, openURICommandId, (IMethod) element, utils);`

Reply:
This handles the null case, so if there is no root-path property, it won't be appended as, localhost:8080/[null]/..

Reply:
Yes I understand the idea, but see my comments at https://github.com/eclipse/lsp4mp/pull/174#discussion_r701420089